app/locustio/common_utils.py

```diff
@@ -288,4 +292,45 @@ def raise_if_login_failed(locust):
         raise exception.StopUser('Action login_and_view_dashboard failed')
 
 
+def run_as_specific_user(username=None, password=None):
+    if not (username and password):
+        raise SystemExit(f'The credentials are not valid: {{username: {username}, password: {password}}}.')
+
+    def deco_wrapper(func):
+        @functools.wraps(func)
+        def wrapper(*args, **kwargs):
+
+            locust = None
+            for obj in list(locals()['kwargs'].values()) + list(locals()['args']):
+                if isinstance(obj, MyBaseTaskSet):
+                    locust = obj
+                    break
+
+            if locust:
+                session_user_name = locust.session_data_storage["username"]
+                session_user_password = locust.session_data_storage["password"]
+                app = locust.session_data_storage['app']
+
+                if app == 'jira' or app == 'jsm':
```

Comment:
It would be nice to move string 'jira' to constant variable JIRA. The same for JSM and CONFLUENCE.

Reply:
good decision, thx